ansible/roles/preflights/tasks/main.yaml

```diff
@@ -0,0 +1,393 @@
+            command: ip addr del {{ keepalived.virtual_ip }} dev {{ keepalived.interface }}
+            when: inventory_hostname == groups['control-plane'][0]
+
+      when:
+        - not keepalived_file.stat.exists
+    when:
+      - keepalived.enabled
+
+  - name: check if /etc/konvoy-marker.yaml exists
+    stat:
+      path: "/etc/konvoy-marker.yaml"
+    register: marker_file
+
+  - name: detect persistent volume disks in /mnt/disks
+    find:
+      paths: '/mnt/disks/'
+      file_type: directory
+    register: pvd
+    when: not marker_file.stat.exists
+
+  - name: check for old persistent volumes in /mnt/disks/*/
+    find:
+      paths: "{{ pvd.files | map(attribute='path') | list }}"
+      file_type: any
+      excludes:
+        - 'lost+found'
+    register: old_pvs
+    failed_when: old_pvs.matched > 0
+    changed_when: old_pvs.matched > 0
+    when:
+      - not marker_file.stat.exists
+      - "'node' in group_names"
+
+  - block:
+    - name: collect OS disk usage on the root volume
+      shell: df -hl | awk '/\/$/ { sum+=$5 } END { print sum }'
+      register: os_root_vol_usage
+
+    - name: ensure disk usage on the root volume is less than {{ preflight.disk_usage_threshold }}%
+      assert:
+        that:
+           - os_root_vol_usage.stdout|int < {{ preflight.disk_usage_threshold }}
+        msg: Disk space has reached {{ preflight.disk_usage_threshold }}% threshold
+
+    - name: collect OS disk allocated size on the root volume
+      shell: df -hl | awk '/\/$/ { sum+=$2 } END { print sum }'
+      register: os_root_vol_size
+
+    - name: ensure disk allocated size on the root volume is equal or greater than {{ preflight.disk_size_threshold }}G
+      ignore_errors: True
+      when:
+        - os_root_vol_size.stdout|int < preflight.disk_size_threshold
+      fail:
+        msg: "WARNING: Disk size {{ os_root_vol_size.stdout|int }}G is less than {{ preflight.disk_size_threshold }}G"
+
+    - name: ensure minimum allocated CPU cores for all the nodes
+      assert:
+        that:
+          - ansible_processor_vcpus >= {{ preflight.minimum_cpu_cores }}
+        msg: CPU core allocation {{ ansible_processor_vcpus }} is less than required {{ preflight.minimum_cpu_cores }}
+
+    - name: ensure minimum allocated memory for control plane nodes
+      assert:
+        that:
+          - ansible_memtotal_mb > {{ preflight.minimum_memory_control_plane_mb|int }}
+        msg: Memory allocation {{ ansible_memtotal_mb }}MB is less than required {{ preflight.minimum_memory_control_plane_mb|int }}MB for control plane nodes
+      when: "'control-plane' in group_names"
+
+    - name: ensure minimum allocated memory for worker nodes
+      assert:
+        that:
+          - ansible_memtotal_mb > {{ preflight.minimum_memory_worker_mb }}
+        msg: Memory allocation {{ ansible_memtotal_mb }}MB is less than required {{ preflight.minimum_memory_worker_mb }}MB for workers
+      when: "'control-plane' not in group_names"
```

Comment:
Do we have `control-plane` and `worker` roles anymore, AFFAIK we agreed to just run all checks on `all` nodes.